localstack/services/cloudwatch/cloudwatch_database_helper.py

```diff
@@ -0,0 +1,407 @@
+
+            query_single_metric = f"""
+            SELECT {STAT_TO_SQLITE_FUNC[stat]} FROM {self.TABLE_SINGLE_METRICS}
+            WHERE account_id = ? AND region = ?
+            AND namespace = ? AND metric_name = ?
+            {dimension_filter}
+            AND timestamp >= ? AND timestamp < ?
+            ORDER BY {order_by}
+            """
+
+            query_agg_metric = f"""
+            SELECT {STAT_TO_SQLITE_COLUMNS[stat]} FROM {self.TABLE_AGGREGATED_METRICS}
+            WHERE account_id = ? AND region = ?
+            AND namespace = ? AND metric_name = ?
+            {dimension_filter}
+            AND timestamp >= ? AND timestamp < ?
+            ORDER BY {order_by}
+            """
+
+            # Execution of query according to period
+            datapoints = {}
+            while start_time_unix < end_time_unix:
+                next_start_time = start_time_unix + period
+                datapoints[str(start_time_unix)] = {"values": []}
+                cur.execute(
+                    query_single_metric,
+                    data
+                    + (
+                        start_time_unix,
+                        next_start_time,
+                    ),
+                )
+                single_result = cur.fetchone()[0]
+                if single_result or (single_result == 0 and stat != "SampleCount"):
+                    datapoints[str(start_time_unix)]["values"].append(single_result)
+
+                cur.execute(
+                    query_agg_metric,
+                    data
+                    + (
+                        start_time_unix,
+                        next_start_time,
+                    ),
+                )
+                agg_result = cur.fetchall()
+                if agg_result:
+                    datapoints[str(start_time_unix)]["values"] += [
+                        (
+                            (
+                                r[0],
+                                r[1],
+                            )
+                            if stat == "Average"
+                            else r[0]
+                        )
+                        for r in agg_result
+                    ]
+
+                start_time_unix = next_start_time
+
+            cleaned_datapoints = {}
+            for timestamp, timestamp_values in datapoints.items():
+                # clean up datapoints
+                cleaned_values = [v for v in timestamp_values["values"] if v is not None]
+                if not cleaned_values:
+                    continue
+
+                # calculate statistic
+                if stat == "Sum" or stat == "SampleCount":
+                    cleaned_datapoints[timestamp] = sum(cleaned_values)
+
+                elif stat == "Minimum":
+                    cleaned_datapoints[timestamp] = min(cleaned_values)
+
+                elif stat == "Maximum":
+                    cleaned_datapoints[timestamp] = max(cleaned_values)
```

Comment:
for sum/min/max we could also use a combined SQL. 
I am still thinking about if we can also make it work for avg 🤔

Reply:
added to follow-up tasks